syft/execution/role_assignments.py

```diff
@@ -10,13 +16,20 @@ class RoleAssignments:
     which other parties are participating and communicate with them if needed.
     """
 
-    def __init__(self, role_ids):
+    def __init__(self, role_ids: list = [], assignments: dict = {}):
```

Comment:
I think there might be a hidden bug when settings obj refs as default arguments (might not occur here, but I would personally avoid it just for future usage).

```
class Test:
    def __init__(self, test={}):
        self.test = test       

    def add_elem(self, k, v):
        self.test[k] = v

    def print(self):
        print(self.test)

t1 = Test()
t1.add_elem("1", 1)
t1.print()
# {"1": 1}

t2 = Test()
t2.add_elem("2", 2)
t2.print()
# {"1": 1, "2": 2}         
```

Reply:
Oh yep, I already had this kind of bug! Nice catch 👍